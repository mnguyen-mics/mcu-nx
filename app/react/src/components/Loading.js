import React from 'react';
import PropTypes from 'prop-types';
import { Spin } from 'antd';

function Loading({ className }) {
  return (
    <div className={`mcs-centered-container ${className}`}>
      <Spin size="large" />
    </div>
  );
}

Loading.defaultProps = {
  className: '',
};

Loading.propTypes = {
  className: PropTypes.string,
};

export default Loading;
