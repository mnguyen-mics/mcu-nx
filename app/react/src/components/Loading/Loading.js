import React, { Component } from 'react';
import { Spin } from 'antd';

class Loading extends Component {
  render() {
    return (
      <div className="mcs-loading-container">
        <Spin size="large" />
      </div>
    );
  }
}

export default Loading;
