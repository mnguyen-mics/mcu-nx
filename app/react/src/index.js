import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
