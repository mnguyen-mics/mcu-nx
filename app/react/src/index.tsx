/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
