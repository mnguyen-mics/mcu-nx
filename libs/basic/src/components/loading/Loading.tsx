import React from 'react';
import { Spin } from 'antd';

export interface LoadingProps {
  className?: string;
  isFullScreen: boolean;
}

const Loading: React.SFC<LoadingProps> = props => {
  return (
    <Spin
      size='large'
      className={`mcs-loading ${props.isFullScreen ? 'loading-full-screen' : ''} ${
        props.className
      }`}
    />
  );
};

export default Loading;
