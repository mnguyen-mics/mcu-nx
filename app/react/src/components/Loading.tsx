import * as React from 'react';
import { Spin } from 'antd';

const Loading: React.SFC<{}> = props => {

  return (
    <div className="mcs-centered-container">
      <Spin size="large" />
    </div>
  );
}

export default Loading;
