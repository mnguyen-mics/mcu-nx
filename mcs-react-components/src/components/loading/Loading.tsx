import * as React from 'react';
import { Spin } from 'antd';

export interface LoadingProps {
  className?: string;
}

const Loading: React.SFC<LoadingProps> = props => {

  return (
    <div className={`mcs-centered-container ${props.className}`}>
      <Spin size="large" />
    </div>
  );
};

export default Loading;
