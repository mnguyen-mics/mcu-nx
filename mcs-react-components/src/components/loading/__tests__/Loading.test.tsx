import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Loading, { LoadingProps } from '../Loading';

it('renders the Loading', () => {
  const props: LoadingProps = {
    isFullScreen: true
  };

  const component = TestRenderer.create(
    <Loading {...props} />
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
