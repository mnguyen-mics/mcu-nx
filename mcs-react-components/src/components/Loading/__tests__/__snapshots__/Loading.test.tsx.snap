// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ColoredButton 1`] = `
<div
  className="mcs-centered-container loading-full-screen"
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i />
      <i />
      <i />
      <i />
    </span>
  </div>
</div>
`;
